src/components/MoneyRequestConfirmationList.tsx

```diff
@@ -248,14 +248,11 @@ function MoneyRequestConfirmationList({
     }, [defaultMileageRate, customUnitRateID, lastSelectedDistanceRates, policy?.id, canUseP2PDistanceRequests, transactionID, isDistanceRequest]);
 
     const policyCurrency = policy?.outputCurrency ?? PolicyUtils.getPersonalPolicy()?.outputCurrency ?? CONST.CURRENCY.USD;
-
     const mileageRate = TransactionUtils.isCustomUnitRateIDForP2P(transaction) ? DistanceRequestUtils.getRateForP2P(policyCurrency) : mileageRates?.[customUnitRateID] ?? defaultMileageRate;
-
-    const {unit, rate} = mileageRate ?? {};
-
+    const unit = DistanceRequestUtils.getDistanceUnit(transaction, mileageRate);
+    const {rate} = mileageRate ?? {};
```

Comment:
We have two sources of truth `mileageRate.unit` and `unit` which could be confusing and error-prone. Can we have the returned `mileageRate` already have the correct unit. i.e. write a function `DistanceRequestUtils.getRate` and in that function pass both the policy and the transaction and set the `unit` accordingly.

Also I think we can remove the `isCustomUnitRateIDForP2P`, `getRateForP2P` and `getCustomUnitRateID` (not sure about this one) functions or at least not export them.  The `getRate` function should be enough

Reply:
That's a pretty good idea, thanks. I'll work on it.

Reply:
I ended up adding two new util functions for getting the rate of existing transactions and then getting the distance unit for use when updating a transaction. 



I left one use of these other utils you recommended removing, because making the current code work with the new utils would require creating a temporary transaction, setting the customUnitRateID from the transaction changes into it, then using one util for the rate and another for the unit. It's just as messy as the existing code. Not to mention that I would also need to change a lot of the OnyxEntryOrInput types to work with the OnyxEntry types. So I'm going to leave that as is.


https://github.com/Expensify/App/blob/7982c2636306b93117144c78288a826ee0d2da76/src/libs/TransactionUtils/index.ts#L807-L809

Please let me know what you think in your next review.